base/pkg/dir.jl

```diff
@@ -29,12 +29,8 @@ function cd(f::Function, args...; kws...)
             throw(PkgError("Package metadata directory $metadata_dir doesn't exist; run Pkg.init() to initialize it."))
     end
     if haskey(ENV,"JULIA_PKGDIR")
-        olddir = ENV["JULIA_PKGDIR"]
-        ENV["JULIA_PKGDIR"] = dir # use absolute path
-        try
+        withenv("JULIA_PKGDIR" => _pkgroot()) do
```

Comment:
I don't think the other one was actually giving double versions, as the `isdir(d) || !isdir(b)` test in `Dir.path()` would have failed. But this is clearer.